### PR TITLE
powerplatform_environment: environment_type change triggers update not recreate

### PR DESCRIPTION
**Pull request by mawasile**
This pull request introduces a new feature to modify the environment type and includes various updates to support this functionality. The changes span multiple files, adding new methods, DTOs, and tests, as well as modifying existing logic to accommodate the new feature.

### New Feature: Modify Environment Type

* [`internal/services/environment/api_environment.go`](diffhunk://#diff-6939b69029f482f476bfc737204e004670b40a16e4a3a6c3af61eb9efb8244afR360-R390): Added `ModifyEnvironmentType` method to allow environment type modification.
* [`internal/services/environment/dto.go`](diffhunk://#diff-29db28f0e3bf93f9d3a7c0ced4207a58b561a2f259cd1c992d4da5cdf91c8462R114-R117): Introduced `modifySkuDto` for the new environment type modification functionality.
* [`internal/services/environment/resource_environment.go`](diffhunk://#diff-f5dfd8b19cc74212734109f64e3404cb525bb247f0786830e21096a996c9e9adL116-L118): Updated `Update` method to call `ModifyEnvironmentType` when the environment type changes. Removed `RequiresReplace` plan modifier for `environment_type`. [[1]](diffhunk://#diff-f5dfd8b19cc74212734109f64e3404cb525bb247f0786830e21096a996c9e9adL116-L118) [[2]](diffhunk://#diff-f5dfd8b19cc74212734109f64e3404cb525bb247f0786830e21096a996c9e9adR414-R421) [[3]](diffhunk://#diff-f5dfd8b19cc74212734109f64e3404cb525bb247f0786830e21096a996c9e9adL493-R504)

### Configuration and Constants

* [`internal/constants/constants.go`](diffhunk://#diff-02820ac94e512b02b1879746ff901479469e1dda3c21277d7f7a170cd1c25c71L95-R96): Reordered constants for better readability.
* [`internal/api/lifecycle.go`](diffhunk://#diff-66f38c9ec36726a5119bb84ea62c468284d58a5e6b6b25ce1a8631f01d0fba56R56-R58): Added fallback to `HEADER_OPERATION_LOCATION` if `HEADER_LOCATION` is not present.

### Testing Enhancements

* [`internal/services/environment/resource_environment_test.go`](diffhunk://#diff-07ad5067382ca0f319e0f77b1f72649a105aa53a05b7231cb67b9249f2bad472R14-R15): Added unit and acceptance tests for validating the new environment type modification feature. [[1]](diffhunk://#diff-07ad5067382ca0f319e0f77b1f72649a105aa53a05b7231cb67b9249f2bad472R14-R15) [[2]](diffhunk://#diff-07ad5067382ca0f319e0f77b1f72649a105aa53a05b7231cb67b9249f2bad472R2016-R2190)
* [`internal/services/environment/tests/resource/Validate_Update_Environment_Type/get_environment_0.json`](diffhunk://#diff-8517bf583a44f7fa1cbfb4d64d2806269c9082f1fec1d276f7a6d72b8533374aR1-R158): Added test data for environment type validation tests.

### CODE REVIEW

File: internal/services/environment/api_environment.go [hunk 1]
```diff
@@ -357,6 +357,37 @@ func (client *Client) AddDataverseToEnvironment(ctx context.Context, environment
...
+	lifecycleResponse, err := client.Api.DoWaitForLifecycleOperationStatus(ctx, apiResponse)
+	if err != nil {
+		return err
+	}
+
+	if lifecycleResponse.State.Id == "Succeeded" {
+		tflog.Debug(ctx, "Environment SKU Update Operation Success")
+		return nil
+	}
+	return errors.New("Environment creation failed. provisioning state: " + lifecycleResponse.State.Id)
```
Comment (mattdot) [inline internal/services/environment/api_environment.go:388]: Reverse these so success falls through to the end and we return errors in ifs.

File: internal/services/environment/resource_environment_test.go [hunk 2]
```diff
@@ -2011,3 +1993,178 @@ func TestAccEnvironmentsResource_Create_Environment_No_Dataverse_Add_Env_Group(t
...
+					resource "powerplatform_environment" "development" {
+					display_name                              = "Example"
+					location                                  = "europe"
+					environment_type                          = "Production"
+					dataverse = {
+						language_code                             = "1033"
+						currency_code                             = "PLN"
+						security_group_id 						  = "00000000-0000-0000-0000-000000000000"
+					}
+				}`,
```
Comment (mattdot) [inline internal/services/environment/resource_environment_test.go:2092]: Should probably test more than just Sandbox >> Production.

**Review by mattdot — CHANGES_REQUESTED** [review]

File: internal/services/environment/resource_environment_test.go (hunk 2 above)
Comment (mawasile) [reply internal/services/environment/resource_environment_test.go:2092]: I think we only need/should test that is changes successfully. I don't think we should test all the configurations and this is up to backend BAPI to work correctly now our code & tests

File: internal/services/environment/api_environment.go (hunk 1 above)
Comment (mawasile) [reply internal/services/environment/api_environment.go:388]: done

**Review by mattdot — APPROVED** [review]

**Pull request merged**